crates/ruff_linter/resources/test/fixtures/flake8_type_checking/TCH008.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+from typing import TypeAlias, TYPE_CHECKING
+
+from foo import Foo
+
+if TYPE_CHECKING:
+    from typing import Dict
+
+    OptStr: TypeAlias = str | None
+    Bar: TypeAlias = Foo[int]
+else:
+    Bar = Foo
+
+a: TypeAlias = 'int'  # TCH008
+b: TypeAlias = 'Dict'  # OK
+c: TypeAlias = 'Foo'   # TCH008
+d: TypeAlias = 'Foo[str]'  # OK
+e: TypeAlias = 'Foo.bar'  # OK
+f: TypeAlias = 'Foo | None'  # TCH008
+g: TypeAlias = 'OptStr'   # OK
+h: TypeAlias = 'Bar'   # TCH008
+i: TypeAlias = Foo['str']   # TCH008
+j: TypeAlias = 'Baz'   # OK
+k: TypeAlias = 'k | None'  # OK
+l: TypeAlias = 'int' | None  # TCH008 (because TC010 is not enabled)
+m: TypeAlias = ('int'  # TCH008
+    | None)
+n: TypeAlias = ('int'  # TCH008 (fix removes comment currently)
+    ' | None')
+
+type B = 'Dict'  # TCH008
+type D = 'Foo[str]'  # TCH008
+type E = 'Foo.bar'  # TCH008
+type G = 'OptStr'  # TCH008
+type I = Foo['str']  # TCH008
+type J = 'Baz'  # TCH008
+type K = 'K | None'  # TCH008
+type L = 'int' | None  # TCH008 (because TC010 is not enabled)
```

Comment:
Indeed, I wasn't quite sure how the ordering of events would play out, so I decided against targeting the 0.8 branch with this change for now. I can clean this up either once the 0.8 branch has been merged or before that if we want to try to get these rules into 0.8.0, rather than 0.8.1+.

I don't think it's worth cleaning up before that decision has been made.